README.md

```diff
@@ -287,6 +500,503 @@ async fn my_function() {
+
+```rust
+#[http(method = "POST")]
+async fn create_user(&mut self, user: User) -> Result<String, String> { ... }
+
+// ❌ Won't match - wrong case
+{"create_user": {...}}
+{"createUser": {...}}
+
+// ✅ Will match - exact CamelCase
+{"CreateUser": {...}}
+```
+
+**Solution:** Always use exact CamelCase conversion: `snake_case` → `CamelCase`
+
+#### **Async Context Limitations:**
+
+```rust
+#[http(method = "POST")]
+async fn async_handler(&mut self, data: MyData) -> String {
+    // ✅ Works - context is preserved by the framework
+    let path = get_path().unwrap_or_default();
+    
+    // ⚠️ Potential issue - long-running tasks might lose context
+    tokio::time::sleep(Duration::from_secs(30)).await;
+    let path2 = get_path(); // May be None if context expires
+    
+    format!("Path: {}", path)
+}
+```
+
+**Why:** The framework preserves context by capturing it before async execution, but very long-running tasks might exceed context lifetime.
+
+#### **Query Parameter Encoding:**
+
+```rust
+// URL: /api/search?q=hello%20world&tags=rust,web
+let params = get_query_params().unwrap();
+
+// ❌ Raw values - URL encoding not automatically decoded
+assert_eq!(params.get("q"), Some("hello%20world")); // Still encoded
+
+// ✅ Manual decoding needed for special characters
+let query = params.get("q")
+    .map(|s| urlencoding::decode(s).unwrap().into_owned())
+    .unwrap_or_default(); // "hello world"
+```
+
+**Solution:** Use a URL decoding library for complex query parameters.
+
+#### **No Built-in Content Negotiation:**
+
+```rust
+#[http(method = "POST")]
+fn upload_file(&mut self, data: FileData) -> String {
+    // ❌ Only handles JSON - no multipart/form-data, XML, etc.
+    // Framework always expects JSON body format
+}
+```
+
+**Workaround:** Use parameter-less handlers and manually parse request bodies:
+
+```rust
+#[http(method = "POST", path = "/upload")]
+fn handle_file_upload(&mut self) -> Result<String, String> {
+    // Get raw request body and parse manually
+    // Implementation depends on your specific needs
+}
+```
+
+#### **Single Handler Per Variant:**
+
+```rust
+// ❌ This won't compile - duplicate handler names become same variant
+#[http(method = "POST")]
+fn create_user(&mut self, user: User) -> User { ... }
+
+#[http(method = "PUT")] 
+fn create_user(&mut self, user: User) -> User { ... } // ERROR: Duplicate CreateUser variant
+```
+
+**Solution:** Use different method names or combine logic:
+
+```rust
+#[http(method = "POST")]
+fn create_user(&mut self, user: User) -> User { ... }
+
+#[http(method = "PUT")]
+fn update_user(&mut self, user: User) -> User { ... }
+
+// Or combine with method checking:
+#[http]
+fn user_handler(&mut self, user: User) -> User {
+    match get_http_method().as_deref() {
+        Some("POST") => self.create_user_impl(user),
+        Some("PUT") => self.update_user_impl(user), 
+        _ => panic!("Unsupported method")
+    }
+}
+```
+
+#### **UI Path Conflicts (Critical!):**
```

Comment:
discussion in this section seems wrong even if we don't remove the ui path hardcoding